src/hotspot/os_cpu/bsd_aarch64/vm_version_bsd_aarch64.cpp

```diff
@@ -24,22 +24,66 @@
  */
 
 #include "precompiled.hpp"
+#include "runtime/java.hpp"
 #include "runtime/os.hpp"
 #include "runtime/vm_version.hpp"
 
+#include <sys/sysctl.h>
+
 int VM_Version::get_current_sve_vector_length() {
-  ShouldNotReachHere();
+  ShouldNotCallThis();
   return -1;
 }
 
 int VM_Version::set_and_get_current_sve_vector_length(int length) {
-  ShouldNotReachHere();
+  ShouldNotCallThis();
   return -1;
 }
 
+static bool cpu_has(const char* optional) {
+  uint32_t val;
+  size_t len = sizeof(val);
+  if (sysctlbyname(optional, &val, &len, NULL, 0)) {
+    return false;
+  }
+  return val;
+}
+
 void VM_Version::get_os_cpu_info() {
-  _icache_line_size = _dcache_line_size = 64;
+  size_t sysctllen;
+
+  // hw.optional.floatingpoint always returns 1.
+  // ID_AA64PFR0_EL1 describes AdvSIMD always equals to FP field.
```

Comment:
Why not keep these tests against `hw.optional.floatingpoint` and `hw.optional.neon`?